homeassistant/components/livisi/switch.py

```diff
@@ -0,0 +1,240 @@
+"""Code to handle a Livisi switches."""
+from aiolivisi import AioLivisi
+
+from homeassistant.components.switch import SwitchEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.entity import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DOMAIN, LOGGER, SWITCH_PLATFORM
+from .device import Device
+from .shc import SHC
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up switch device."""
+
+    shc = hass.data[DOMAIN][config_entry.entry_id]
+
+    @callback
+    async def async_add_switch(device) -> None:
+        """Add switch."""
+        livisi_switch: LivisiSwitch = await create_device(device, hass, shc)
+        LOGGER.debug("Include device type: %s", device.get("type"))
+        shc.included_devices.append(livisi_switch)
+        async_add_entities([livisi_switch])
+
+    shc.register_new_device_callback(SWITCH_PLATFORM, async_add_switch)
+    await async_setup_switch_entry(hass, shc, async_add_entities)
+
+
+async def async_setup_switch_entry(
+    hass: HomeAssistant,
+    shc: SHC,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up switch device."""
+
+    livisi_switches = []
+    for device in shc.switch_devices:
+        if device in shc.included_devices:
+            continue
+        livisi_switch: LivisiSwitch = await create_device(device, hass, shc)
+        livisi_switches.append(livisi_switch)
+        LOGGER.debug("Include device type: %s", device.get("type"))
+        shc.included_devices.append(livisi_switch)
```

Comment:
Don't store entities in a shared container. Entities are not allowed to be accessed from outside. Use our dispatch helper or our update coordinator helper to tell entities to update state from new data.